sw/device/tests/chip_sw_ast_clk_rst_inputs.c

```diff
@@ -0,0 +1,498 @@
+  {
+    if(read_fifo_depth_val>=read_fifo_depth(&entropy_src))
+      LOG_ERROR("read_fifo_depth after exit from idle=%0d should be equal/greater than previous read value (%0d)",read_fifo_depth(&entropy_src), read_fifo_depth_val);
+  }
+
+  IBEX_SPIN_FOR(read_fifo_depth(&entropy_src) > 0, 1000);
+
+  if(test_idx==kAstAlertAfterLp)
+  {
+    // test recoverable event
+    test_event(alert_idx, kDifToggleDisabled,false);
+  }
+  
+  CHECK_DIF_OK(dif_pwrmgr_wakeup_reason_clear(&pwrmgr)); 
+
+  // test event after exit from low power
+  if(test_idx==kAstAlertAfterLp)
+  {
+  test_event(alert_idx, kDifToggleDisabled,true);
+  test_event(alert_idx, kDifToggleDisabled,false);
+  }                            
+}
+
+
+
+
+void set_edn_auto_mode()
+{
+
+  const dif_edn_t edn0 = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR)};
+  const dif_edn_t edn1 = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_EDN1_BASE_ADDR)};
+
+  //entropy_testutils_stop_all();
+  CHECK_DIF_OK(dif_edn_stop(&edn0));
+  CHECK_DIF_OK(dif_edn_stop(&edn1));
+
+  // re-eanble entropy src and csrng
+  //setup_entropy_src(&entropy_src);
+  //CHECK_DIF_OK(dif_csrng_configure(&csrng));
+
+  // Re-enable EDN0 in auto mode.
+  const dif_edn_auto_params_t edn0_params = {
+      // EDN0 provides lower-quality entropy.  Let one generate command return 8
+      // blocks, and reseed every 32 generates.
+      .instantiate_cmd =
+          {
+              .cmd = 0x00000001 |  // Reseed from entropy source only.
+                     kMultiBitBool4False << 8,
+              .seed_material =
+                  {
+                      .len = 0,
+                  },
+          },
+      .reseed_cmd =
+          {
+              .cmd = 0x00008002 |  // One generate returns 8 blocks, reseed
+                                   // from entropy source only.
+                     kMultiBitBool4False << 8,
+              .seed_material =
+                  {
+                      .len = 0,
+                  },
+          },
+      .generate_cmd =
+          {
+              .cmd = 0x00008003,  // One generate returns 8 blocks.
+              .seed_material =
+                  {
+                      .len = 0,
+                  },
+          },
+      .reseed_interval = 32,  // Reseed every 32 generates.
+  };
+  CHECK_DIF_OK(dif_edn_set_auto_mode(&edn0, edn0_params));
+
+  // Re-enable EDN1 in auto mode.
+  const dif_edn_auto_params_t edn1_params = {
+      // EDN1 provides highest-quality entropy.  Let one generate command
+      // return 1 block, and reseed after every generate.
+      .instantiate_cmd =
+          {
+              .cmd = 0x00000001 |  // Reseed from entropy source only.
+                     kMultiBitBool4False << 8,
+              .seed_material =
+                  {
+                      .len = 0,
+                  },
+          },
+      .reseed_cmd =
+          {
+              .cmd = 0x00001002 |  // One generate returns 1 block, reseed
```

Comment:
Is there a way to express 0x1002 with symbols to make it more obvious what this means? Same above and below.